addons/payment/models/payment_acquirer.py

```diff
@@ -227,6 +227,36 @@ def _get_default_payment_method_id(self):
         self.ensure_one()
         return self.env.ref('account.account_payment_method_manual_in').id
 
+    #=== ONCHANGE METHODS ===#
+
+    @api.onchange('state')
+    def _onchange_state(self):
+        """ Display a warning about the consequences of disabling an acquirer. """
+        self.ensure_one()
+
+        if self.state == 'disabled' and self.id:
+            related_tokens = self.env['payment.token'].search([
+                ('acquirer_id', '=', self._origin.id)
+            ])
+            if (related_tokens):
+                if len(related_tokens) == 1:
+                    message = _(
+                        "Disabling this acquirer will also archive one token registered "
+                        "with this acquirer. Archiving tokens is irreversible."
+                    )
+                else:
+                    message = _(
+                        "Disabling this acquirer will also archive %s tokens registered "
+                        "with this acquirer. Archiving tokens is irreversible.", len(related_tokens)
+                    )
```

Comment:
I think we could go for a generic message `%s token(s)` to have one less translation to manage in database.

Reply:
Well, considering the guidelines (https://www.odoo.com/documentation/master/developer/howtos/translations.html?highlight=plural#plural), let's keep it like this then.

Ty @chevalierv :D